plugin/etcd/cname_test.go

```diff
@@ -83,4 +84,11 @@ var dnsTestCasesCname = []test.Case{
 			test.CNAME("region2.skydns.test.	300	IN	CNAME	mainendpoint.region2.skydns.test."),
 		},
 	},
+	{
+		Qname: "region3.skydns.test.", Qtype: dns.TypeCNAME,
+		Rcode: dns.RcodeNameError,
```

Comment:
For this test it needs to return Rcode: dns.RcodeSuccess with (still) the SOA in the Auth section.

Reply:
Hmmm. Yes, you're right of course, since there actually is a record (TXT) for the queried domain. I got hung up on the fact that the entry that was requested didn't exist for the requested type.